src/libraries/System.Private.CoreLib/src/System/Decimal.cs

```diff
@@ -1150,15 +1150,39 @@ object IConvertible.ToType(Type type, IFormatProvider? provider)
         // IFloatingPoint
         //
 
+        /// <inheritdoc cref="IFloatingPoint{TSelf}.GetExponentByteCount()" />
+        int IFloatingPoint<decimal>.GetExponentByteCount() => sizeof(sbyte);
+
         /// <inheritdoc cref="IFloatingPoint{TSelf}.GetExponentShortestBitLength()" />
-        long IFloatingPoint<decimal>.GetExponentShortestBitLength()
+        int IFloatingPoint<decimal>.GetExponentShortestBitLength()
         {
             sbyte exponent = Exponent;
             return (sizeof(sbyte) * 8) - sbyte.LeadingZeroCount(exponent);
         }
 
-        /// <inheritdoc cref="IFloatingPoint{TSelf}.GetExponentByteCount()" />
-        int IFloatingPoint<decimal>.GetExponentByteCount() => sizeof(sbyte);
+        /// <inheritdoc cref="IFloatingPoint{TSelf}.GetSignificandByteCount()" />
+        int IFloatingPoint<decimal>.GetSignificandByteCount() => sizeof(ulong) + sizeof(uint);
+
+        /// <inheritdoc cref="IFloatingPoint{TSelf}.GetSignificandBitLength()" />
+        int IFloatingPoint<decimal>.GetSignificandBitLength() => 96;
```

Comment:
It seems weird to me that ByteCount uses sizeof and this uses a literal.  The compiler should be able to fold `(sizeof(ulong) + sizeof(uint)) * 8`, and then the magic number has an explanation.

Alternatively, just `GetSignificandByteCount() * 8`, and let the JIT fold it.

Reply:
If returning an Int128 here is a thing you might actually want to do, I recommend expanding to a 16 byte response now, as it’s a breaking change later.

Reply:
It wouldn't return an `Int128`, `GetSignificandByteCount()` could return `sizeof(Int128)` and therefore write `16-bytes` as an optimization.

This would not be a breaking change because there is no guarantee that `GetSignificandByteCount()` returns a specific value. Just as `T.GetByteCount()` may return different values over time for `BigInteger` if its underlying implementation were to change.

Reply:
But until `Int128` exists, returning `12-bytes` is "better" since that exactly contains the significand and is just as easy (or difficult) to work with as 2x independent `ulong`

Reply:
> It wouldn't return an Int128, GetSignificandByteCount() could return sizeof(Int128) and therefore write 16-bytes as an optimization.

I think it'd be enough of a breaking change that we'd want to write it into a breaking change doc.  Someone with the current implementation would always see success if they `stackalloc(12)` and call `((IFloatingPoint<decimal>)val).WriteSignificandWhicheverEndian(dest)`, and that would turn into an exception if the minimum is changed to 16 bytes.  While it doesn't break the interface's implied contract it would break libraries that were implemented using .NET N and then get run on .NET N+1(+).

Of course, it could be done in a non-breaking manner, where if destination >= 16 write it one way, else if >= 12 write it another.

Reply:
> That code isn't valid (you can't access static abstract members via casting; only via a constrained generic).

Yeah, but this one is just an explicitly implemented instance method, right?  (I'm not trying to further debate the breakingness of such a change, just making sure that there's not something fundamental I'm misunderstanding about new interface syntaxes)

Reply:
Ah right, its an instance method so the cast is valid.

I keep forgetting that the `GetByteCount` and `TryWrite` APIs are the exception, and are such since we need them so devs can use them in the `Create` APIs.

---------

I do plan on updating this to be `16-bytes` for .NET 7 once https://github.com/dotnet/runtime/pull/69204 is merged, but I would still want to hold the line on not quantifying user bugs in API usage as breaking changes.